bin/update-readmes.js

```diff
@@ -1,70 +1,138 @@
 /**
- * Node dependencies.
+ * External dependencies
  */
-const { join } = require( 'path' );
-const spawnSync = require( 'child_process' ).spawnSync;
-
-const packages = [
-	'a11y',
-	'autop',
-	'blob',
-	'block-editor',
-	'block-library',
-	'block-serialization-default-parser',
-	'blocks',
-	'compose',
-	[ 'core-data', {
-		'Autogenerated actions': 'src/actions.js',
-		'Autogenerated selectors': 'src/selectors.js',
-	} ],
-	'data',
-	'data-controls',
-	'date',
-	'deprecated',
-	'dom',
-	'dom-ready',
-	'e2e-test-utils',
-	'edit-post',
-	'element',
-	'escape-html',
-	'html-entities',
-	'i18n',
-	'keycodes',
-	'plugins',
-	'priority-queue',
-	'redux-routine',
-	'rich-text',
-	'shortcode',
-	'url',
-	'viewport',
-	'wordcount',
-];
-
-packages.forEach( ( entry ) => {
-	if ( ! Array.isArray( entry ) ) {
-		entry = [ entry, { 'Autogenerated API docs': 'src/index.js' } ];
+const { join, relative, resolve, sep } = require( 'path' );
+const glob = require( 'fast-glob' );
+const execa = require( 'execa' );
+const { Transform } = require( 'stream' );
+const { readFile } = require( 'fs' ).promises;
+
+/**
+ * README file tokens, defined as a tuple of token identifier, source path.
+ *
+ * @typedef {[string,string]} WPReadmeFileTokens
+ */
+
+/**
+ * README file data, defined as a tuple of README file path, token details.
+ *
+ * @typedef {[string,WPReadmeFileTokens]} WPReadmeFileData
+ */
+
+/**
+ * Path to packages directory.
+ *
+ * @type {string}
+ */
+const PACKAGES_DIR = resolve( __dirname, '../packages' );
+
+/**
+ * Pattern matching start token of a README file.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = /<!-- START TOKEN\((.+?(?:\|(.+?))?)\) -->/g;
+
+/**
+ * Given an absolute file path, returns the package name.
+ *
+ * @param {string} file Absolute path.
+ *
+ * @return {string} Package name.
+ */
+function getFilePackage( file ) {
+	return relative( PACKAGES_DIR, file ).split( sep )[ 0 ];
+}
+
+/**
+ * Returns an appropriate glob pattern for the packages directory to match
+ * relevant for a given set of files.
+ *
+ * @param {string[]} files Set of files to match. Pass an empty set to match
+ *                         all packages.
+ *
+ * @return {string} Packages glob pattern.
+ */
+function getPackagePattern( files ) {
+	if ( ! files.length ) {
+		return '*';
 	}
 
-	const [ packageName, targets ] = entry;
-
-	Object.entries( targets ).forEach( ( [ token, path ] ) => {
-		// Each target operates over the same file, so it needs to be processed synchronously,
-		// as to make sure the processes don't overwrite each other.
-		const { status, stderr } = spawnSync(
-			join( __dirname, '..', 'node_modules', '.bin', 'docgen' ).replace( / /g, '\\ ' ),
```

Comment:
Note: The changes here remove the `String#replace` fix which was introduced by @MarkMarzeotti in #18253 to improve Windows support.

My reading of the documentation of `execa` leads me to believe that this should be handled by default in the library, as well as potential other issues we could have with Windows support:

>Node has issues when using spawn on Windows:
>
>[...]
>- Has problems running commands with [spaces](https://github.com/nodejs/node/issues/7367)

https://github.com/moxystudio/node-cross-spawn#why